openff/toolkit/typing/engines/smirnoff/forcefield.py

```diff
@@ -1505,6 +1505,34 @@ def _get_parameter_handler_class(self, tagname):
             raise KeyError(msg)
         return ph_class
 
+    def get_partial_charges(self, molecule, **kwargs):
+        """Generate the partial charges for the given molecule in this force field.
+
+        As any `ParameterHandler` may in principle modify charges, the entire
+        force field must be applied to the molecule to produce the charges.
+        Calls to this method from incorrectly or incompletely specified `ForceField`
+        objects thus may raise an exception.
+
+        Parameters
+        ----------
+        molecule : openff.toolkit.topology.Molecule
+            The ``Molecule`` corresponding to the system to be parameterized
+        toolkit_registry : openff.toolkit.utils.toolkits.ToolkitRegistry, optional. default=GLOBAL_TOOLKIT_REGISTRY
+            The toolkit registry to use for operations like conformer generation and
+            partial charge assignment.
+
+        Returns
+        -------
+        charges : simtk.unit.Quantity with shape (n_atoms,) with units compatible with elemental charge
+            The partial charges of the provided molecule in this force field.
+
+        """
+        _, top_with_charges = self.create_openmm_system(
+            molecule.to_topology(), return_topology=True, **kwargs
+        )
+        charges = [*top_with_charges.reference_molecules][0].partial_charges
```

Comment:
We should do a check for whether this molecule has vsites `offmol.n_virtual_particles`. If it DOES, I'd be ok with raising an error, since the vsites may steal charge from the atoms, and it won't be clear to the user whether they're getting charges BEFORE or AFTER this adjustment.

Reply:
If you add the following OFFXML to the `test_forcefield`, you can make a test that will assign a virtualsite to ethanol

https://github.com/openforcefield/openff-toolkit/blob/18ea825c8598d02268f5607452116adb63ca6a26/openff/toolkit/tests/test_forcefield.py#L1976-L1995